NEXT-12479 - Add documentation for new Stock API (#996)
shopware/docs: guides/plugins/plugins/content/stock/loading-stock-information-from-different-source.md

# Loading Stock Information from a Different Source

## Overview

If Shopware is not the source of truth for your stock data, you can customize the stock loading process and provide your data from a third-party source.

## Prerequisites

> We will be decorating a service; therefore, familiarising yourself with the [Adjusting a Service](../../plugin-fundamentals/adjusting-service) guide will be helpful.

Reviewer: ```suggestion
Here again, you will be decorating a service; therefore, it will be helpful to familiarise yourself with the [Adjusting a Service](../../../../../plugin-fundamentals/adjusting-service) guide.
```

> Here again, you will be decorating a service; therefore, it will be helpful to familiarise yourself with the [Adjusting a Service](../../../../../plugin-fundamentals/adjusting-service) guide.

Reviewer: ```suggestion
Here again, you will be decorating a service; therefore, it will be helpful to familiarise yourself with the [Adjusting a Service](../../../../../guides/plugins/plugins/plugin-fundamentals/adjusting-service.md) guide.
```

## Add a decorator to load the stock

> For example, we want to load stock from a third-party API. We will decorate `\Shopware\Core\Content\Product\Stock\AbstractStockStorage` and implement the `load` method. When products are loaded in Shopware the `load` method will be invoked with the loaded product IDs.

Reviewer: ```suggestion
For example, to load stock from a third-party API, you need to decorate `\Shopware\Core\Content\Product\Stock\AbstractStockStorage` and implement the `load` method. When products are loaded in Shopware the `load` method will be invoked with the loaded product IDs.
```

| Attribute   | Type    | Description                                                     | Optional/Required |
|-------------|---------|-----------------------------------------------------------------|-------------------|
| productId   | string  | The product ID                                                  | Required          |
| stock       | int     | The stock amount                                                | Required          |
| available   | boolean | Whether the product is considered available                     | Required          |
| minPurchase | int     | The minimum purchase value for this product                     | Optional          |
| maxPurchase | int     | The maximum purchase value for this product                     | Optional          |
| isCloseout  | boolean | Whether the product can be ordered if there is not enough stock | Optional          |

For example:

```php
$stockData = \Shopware\Core\Content\Product\Stock\StockData::fromArray([
    'productId' => 'product-1',
    'stock' => 5,
    'available' => true,
    'minPurchase' => 1,
    'maxPurchase' => 10,
    'isCloseout' => false,
]);
```

> It's also possible to provide arbitrary data via extensions:

Reviewer: ```suggestion
It is also possible to provide arbitrary data via extensions:
```
